iree/compiler/Dialect/HAL/Target/LLVM/internal/UnixLinkerTool.cpp

```diff
@@ -30,30 +32,32 @@ class UnixLinkerTool : public LinkerTool {
   using LinkerTool::LinkerTool;
 
   std::string getToolPath() const override {
+    // Ideally a user specifies the linker so they can match whatever they are
+    // building with.
     auto toolPath = LinkerTool::getToolPath();
     if (!toolPath.empty()) return toolPath;
 
-// Attempt to find a linker on the path. Ideally a user specifies the linker so
-// they can match whatever they are building with.
-//
-// TODO(ataei): support discovery of linkers on Windows/etc; popen is
-// linux-only. There may be infra in LLVM to do this kind of stuff in a
-// cross-platform way.
-#if !defined(WIN32)
-#define UNIX_SYS_LINKER_PATH_LENGTH 255
-    auto sysLinkers = {"ld", "ld.gold", "lld.ld"};
-    for (auto syslinker : sysLinkers) {
-      FILE *pipe =
-          popen(llvm::Twine("which ").concat(syslinker).str().c_str(), "r");
-      char linkerPath[UNIX_SYS_LINKER_PATH_LENGTH];
-      if (fgets(linkerPath, sizeof(linkerPath), pipe) != NULL) {
-        return strtok(linkerPath, "\n");
+    // If no linker was specified, attempt to find one.
+    auto sysLinkers = {"ld", "ld.gold", "ld.lld"};
```

Comment:
May need to append `.exe` file extension on Windows (LLVM has utils for this)

Reply:
Could also move the lookup code to the base class. It will be helpful for finding other tools (e.g. wasm-ld)